### PR TITLE
StringToByte without mem-allocation supports v1.20

### PR DESCRIPTION
**Pull request by tylitianrui**
### Type of change:

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches
- [ ] Documentation
- [ ] Refactor
- [x] Chore
- [ ] CI/CD or Tests

### What this PR does / why we need it:
Since Go 1.20, the [(Slice|String)Header types will be depreciated](https://github.com/golang/go/commit/bc2e961cf4cda945dc6d607a5076b4e376553e9f) and not recommended to be used.  
 Two functions, unsafe.StringData and unsafe.SliceData will  take over the use cases of two old reflect types.

### Pre-submission checklist:

- [X] Did you explain what problem does this PR solve? Or what new features have been added?
- [X] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by tao12345666333** [comment]
Thanks for your contribution.

I think we should upgrade our Go tool chain to v1.20 first.

File: pkg/utils/s2b_new.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// limitations under the License.
+
+//go:build go1.20
+// +build go1.20
+
+package utils
+
+import "unsafe"
+
+// s2b converts string to a byte slice without memory allocation.
```
Comment (tao12345666333) [inline pkg/utils/s2b_new.go:23]:
```suggestion
// String2Byte converts string to a byte slice without memory allocation.
```

**Review by tao12345666333 — COMMENTED** [review]
Thanks for your contribution. We have now upgraded the Go toolchain for this project to v1.20. Can you merge the latest code from the master branch?

File: pkg/utils/s2b_new.go (hunk 1 above)
Comment (tylitianrui) [reply pkg/utils/s2b_new.go:23]: i finished it

**Review by tao12345666333 — APPROVED** [review]

**Pull request merged**